File: array/find_missing_number.js
// array should be sorted

const findMissingNumber = (arr) => {
  const missingValues = [];
  for (var left = 0; left < arr.length - 1; left++) {
    let currentValue = arr[left];
    let nextValue = arr[left + 1];

    while (nextValue - currentValue > 1) {
      currentValue++;
      missingValues.push(currentValue);
    }
  }
  return missingValues;
};

console.log(findMissingNumber([-10,1, 3, 4, 5, 8]));

File: array/find_missing_number.py
# array should be sorted

def findMissingNumber(arr: list[int]):
    missing_values = []
    for i in range(len(arr)-1):
        current_value = arr[i]
        next_value = arr[i + 1]
        while next_value - current_value > 1:
            current_value = current_value + 1
            missing_values.append(current_value)

    return missing_values


print(findMissingNumber([-10,1, 3, 4, 5, 8]))
